compiler/noirc_frontend/src/hir/resolution/resolver.rs

```diff
@@ -193,16 +193,55 @@ impl<'a> Resolver<'a> {
     /// Since lowering would require scope data, unless we add an extra resolution field to the AST
     pub fn resolve_function(
         mut self,
-        func: NoirFunction,
+        mut func: NoirFunction,
         func_id: FuncId,
     ) -> (HirFunction, FuncMeta, Vec<ResolverError>) {
         self.scopes.start_function();
         self.current_item = Some(DependencyId::Function(func_id));
 
         // Check whether the function has globals in the local module and add them to the scope
         self.resolve_local_globals();
-
         self.add_generics(&func.def.generics);
+
+        // TODO: better way to get a unique generic/global ident?
```

Comment:
Noted that this is still in draft but would be nice to pull this out into `desugar_impl_trait_args` helper